homeassistant/components/rest/sensor.py

```diff
@@ -191,12 +201,24 @@ def update(self):
 
         self.rest.update()
         value = self.rest.data
+        content_type = self.rest.headers.get("content-type")
+
+        if content_type and content_type.startswith("text/xml"):
+            try:
+                value = json.dumps(xmltodict.parse(value))
+            except ExpatError:
+                _LOGGER.warning(
+                    "REST xml result could not be parsed and converted to JSON."
+                )
+                _LOGGER.debug("Erroneous XML: %s", value)
 
         if self._json_attrs:
             self._attributes = {}
             if value:
                 try:
                     json_dict = json.loads(value)
+                    if self._json_attrs_path is not None:
+                        json_dict = jsonpath(json_dict, self._json_attrs_path)
                     if isinstance(json_dict, list):
```

Comment:
This might be unexpected if a JSON path is given, so let's only do it if its omitted.
```suggestion
                    elif isinstance(json_dict, list):
```

Reply:
That will likely be confusing for some content.  Will switch to elif

Reply:
jsonpath actually stores the result in json_dict[0] for what appears to be compatibility with the original js version.   https://jsonpath.com/ shows the same behavior

Reply:
I added a comment to this effect as it definitely isn't something you would expect

Reply:
Weird. but ok.